sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusRetryPolicy.cs

```diff
@@ -222,12 +222,17 @@ private async Task ScheduleResetServerBusy(CancellationToken cancellationToken)
             try
             {
                 await Task.Delay(ServerBusyBaseSleepTime, cancellationToken).ConfigureAwait(false);
-                ResetServerBusy();
             }
             catch (OperationCanceledException)
```

Comment:
Do we intentionally want to throw for other exceptions (although seems very rare)? Since this task is un-awaited, could that cause an unobserved exception error?

Reply:
Fair question.  Since it looks like it is possible for the cancellation token to trigger a disposed exception, I added a general catch block as well.